akka-actor/src/main/scala/akka/io/SimpleDnsCache.scala

```diff
@@ -17,39 +24,71 @@ private[io] trait PeriodicCacheCleanup {
   def cleanup(): Unit
 }
 
-class SimpleDnsCache extends Dns with PeriodicCacheCleanup {
+class SimpleDnsCache extends Dns with PeriodicCacheCleanup with NoSerializationVerificationNeeded {
   import SimpleDnsCache._
-
-  private val cache = new AtomicReference(
-    new Cache[String, Dns.Resolved](immutable.SortedSet()(expiryEntryOrdering[String]()), Map(), () => clock))
+  private val cacheRef = new AtomicReference(
+    new Cache[(String, RequestType), Resolved](
+      immutable.SortedSet()(expiryEntryOrdering()),
+      immutable.Map(),
+      () => clock))
 
   private val nanoBase = System.nanoTime()
 
-  override def cached(name: String): Option[Resolved] = {
-    cache.get().get(name)
+  /**
+   * Gets any IPv4 and IPv6 cached entries.
+   * To get Srv or just one type use DnsProtocol
+   *
+   * This method is deprecated and involves a copy from the new protocol to
+   * remain compatable
+   */
+  @silent("deprecated")
+  override def cached(name: String): Option[Dns.Resolved] = {
+    // adapt response to the old protocol
+    val ipv4 = cacheRef.get().get((name, Ip(ipv6 = false))).toList.flatMap(_.records)
+    val ipv6 = cacheRef.get().get((name, Ip(ipv4 = false))).toList.flatMap(_.records)
+    val both = cacheRef.get().get((name, Ip())).toList.flatMap(_.records)
+
+    println("ipv4: " + ipv4)
+    println("ipv6: " + ipv6)
+    println("both: " + both)
+
+    val all = (ipv4 ++ ipv6 ++ both).collect {
+      case r: ARecord    => r.ip
+      case r: AAAARecord => r.ip
+    }
```

Comment:
perf improvement probably possible here but I guess these are not very often long lists?

Reply:
And it is in the deprecated path that akka does not use